src/include/daos_pipeline.h

```diff
@@ -0,0 +1,261 @@
+/**
+ * (C) Copyright 2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#ifndef __DAOS_PIPE_H__
+#define __DAOS_PIPE_H__
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+
+/** logic functions */
+#define DAOS_FILTER_FUNC_EQ		0
+#define DAOS_FILTER_FUNC_NE		1
+#define DAOS_FILTER_FUNC_LT		2
+#define DAOS_FILTER_FUNC_LE		3
+#define DAOS_FILTER_FUNC_GE		4
+#define DAOS_FILTER_FUNC_GT		5
+#define DAOS_FILTER_FUNC_LIKE		6
+#define DAOS_FILTER_FUNC_ISNULL		7
+#define DAOS_FILTER_FUNC_ISNOTNULL	8
+#define DAOS_FILTER_FUNC_AND		9
+#define DAOS_FILTER_FUNC_OR		10
+
+/** aggregation functions */
+#define DAOS_FILTER_FUNC_SUM		100
+#define DAOS_FILTER_FUNC_MIN		101
+#define DAOS_FILTER_FUNC_MAX		102
+#define DAOS_FILTER_FUNC_AVG		103
+
+/** keys, constants */
+#define DAOS_FILTER_DKEY		200
+#define DAOS_FILTER_AKEY		201
+#define DAOS_FILTER_CONST		202
+
+/** types used in a filter object */
+#define DAOS_FILTER_TYPE_BINARY		0
+#define DAOS_FILTER_TYPE_STRING		1
+#define DAOS_FILTER_TYPE_INTEGER	2
+#define DAOS_FILTER_TYPE_REAL		3
+
+/** types of pipeline nodes */
+#define DAOS_PIPELINE_CONDITION		0
+#define DAOS_PIPELINE_AGGREGATION	1
```

Comment:
I would rather use string instead of defines to identify the type of operation that need to be performed. This would thus be easier to extend in dynamically in the future.

Reply:
@johannlombardi We can do that!